pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -600,7 +610,28 @@ func (exp *FlowExporter) findFlowType(conn flowexporter.Connection) uint8 {
 	return 0
 }
 
+func (exp *FlowExporter) fillEgressPolicyInfo(conn *flowexporter.Connection) {
+	egressPolicyName, err := exp.egressQuerier.GetEgress(conn.SourcePodNamespace, conn.SourcePodName)
+	if err != nil {
+		return
+	}
+	conn.EgressPolicyName = egressPolicyName
+	egressIP, err := exp.egressQuerier.GetEgressIPByName(conn.EgressPolicyName)
+	if err != nil {
+		klog.ErrorS(err, "error when try to get Egress IP by Egress Policy Name", "Egress Policy", conn.EgressPolicyName)
+	}
+	conn.EgressIP = egressIP
+	klog.V(4).InfoS("Fill Egress Policy Info for flow", "Egress Policy", conn.EgressPolicyName, "Egress IP", conn.EgressIP, "Source Pod Namespace", conn.SourcePodNamespace, "Source Pod Name", conn.SourcePodName)
+}
+
 func (exp *FlowExporter) exportConn(conn *flowexporter.Connection) error {
+	conn.FlowType = exp.findFlowType(*conn)
+	if conn.FlowType == ipfixregistry.FlowTypeToExternal && conn.SourcePodNamespace != "" && conn.SourcePodName != "" {
+		// Check if Egress Controller is available
+		if egressController, ok := exp.egressQuerier.(*egress.EgressController); ok && egressController != nil {
```

Comment:
If you have to check for the underlying concrete type (`*egress.EgressController`), it kind of defeats the purpose of using an interface (`egress.EgressQuerier`) in the first place, since now this package has to be aware of that underlying type.

Instead, you have to choices:
- pass a boolean value to `NewFlowExporter` to indicate whether the Egress feature is enabled; that means one more function parameter...
- use a `nil` interface value (which you can just test with `exp.egressQuerier == nil`) to `NewFlowExporter`; code will be slightly more verbose in `cmd/antrea-agent`, but less verbose here